ext/strings_test.go

```diff
@@ -329,3 +330,265 @@ func TestStringsWithExtension(t *testing.T) {
+			format:         "%H is 20 in hexadecimal",
+			formatArgs:     "30",
+			expectedOutput: "1E is 20 in hexadecimal",
+		},
+		{
+			name:           "string support with hexadecimal formatting clause",
+			format:         "%h",
+			formatArgs:     `"Hello world!"`,
+			expectedOutput: "48656c6c6f20776f726c6421",
+		},
+		{
+			name:           "string support with uppercase hexadecimal formatting clause",
+			format:         "%H",
+			formatArgs:     `"Hello world!"`,
+			expectedOutput: "48656C6C6F20776F726C6421",
+		},
+		{
+			name:           "byte support with hexadecimal formatting clause",
+			format:         "%h",
+			formatArgs:     `b"byte string"`,
+			expectedOutput: "6279746520737472696e67",
+		},
+		{
+			name:           "byte support with uppercase hexadecimal formatting clause",
+			format:         "%H",
+			formatArgs:     `b"byte string"`,
+			expectedOutput: "6279746520737472696E67",
+		},
+		{
+			name:           "lowercase scientific notation formatting clause",
+			format:         "%.6e",
+			formatArgs:     "1052.032911275",
+			expectedOutput: "1.052033\u202f\u00d7\u202f10\u2070\u00b3",
+		},
+		{
+			name:           "uppercase scientific notation formatting clause",
+			format:         "%.6E",
+			formatArgs:     "1052.032911275",
+			expectedOutput: "1.052033\u202f\u00d7\u202f10\u2070\u00b3",
+		},
+		{
+			name:           "locale support",
+			format:         "%.3f",
+			formatArgs:     "3.14",
+			locale:         "fr_FR",
+			expectedOutput: "3,140",
+		},
+		{
+			name:           "default precision for fixed-point clause",
+			format:         "%f",
+			formatArgs:     "2.71828",
+			expectedOutput: "2.718280",
+		},
+		{
+			name:           "default precision for scientific notation",
+			format:         "%e",
+			formatArgs:     "2.71828",
+			expectedOutput: "2.718280\u202f\u00d7\u202f10\u2070\u2070",
+		},
+		{
+			name:           "unicode output for scientific notation",
+			format:         "unescaped unicode: %e, escaped unicode: %e",
+			formatArgs:     "2.71828, 2.71828",
+			expectedOutput: "unescaped unicode: 2.718280 × 10⁰⁰, escaped unicode: 2.718280\u202f\u00d7\u202f10\u2070\u2070",
+		},
+		{
+			name:           "NaN support for fixed-point",
+			format:         "%f",
+			formatArgs:     `"NaN"`,
+			expectedOutput: "NaN",
+		},
+		{
+			name:           "positive infinity support for fixed-point",
+			format:         "%f",
+			formatArgs:     `"Infinity"`,
+			expectedOutput: "∞",
+		},
+		{
+			name:           "negative infinity support for fixed-point",
+			format:         "%f",
+			formatArgs:     `"-Infinity"`,
+			expectedOutput: "-∞",
+		},
+		{
+			name:           "timestamp support for string",
+			format:         "%s",
+			formatArgs:     `timestamp("2023-02-03T23:31:20+00:00")`,
+			expectedOutput: "2023-02-03T23:31:20Z",
+		},
+		{
+			name:           "duration support for string",
+			format:         "%s",
+			formatArgs:     `duration("1h45m47s")`,
+			expectedOutput: "6347s",
```

Comment:
I was expecting the opposite. For `duration("6347s")` I was expecting `"1h45m47s"` to be output.

Reply:
Right - `Duration` now uses the human-readable string instead.

Reply:
The expected output looks wrong here?

Reply:
nvm, I see the other comment thread about %s handling.